### ISSUE TITLE
Azure endpoint 

### ISSUE THREAD

**Issue by fbrand-new** [D7EAD/liboai#31]
### Describe the feature or improvement you're requesting

Microsoft azure provides a REST API very similar to the one employed by OpenAI for the same service. Would this be of any interest for the project? 

### Additional context

For example, for chat completion api, Azure looks like 
```curl
curl $AZURE_OPENAI_ENDPOINT/openai/deployments/gpt-35-turbo/chat/completions?api-version=2023-05-15 \
  -H "Content-Type: application/json" \
  -H "api-key: $AZURE_OPENAI_KEY" \
  -d '{"messages":[{"role": "system", "content": "You are a helpful assistant."},{"role": "user", "content": "Does Azure OpenAI support customer managed keys?"},{"role": "assistant", "content": "Yes, customer managed keys are supported by Azure OpenAI."},{"role": "user", "content": "Do other Azure Cognitive Services support this too?"}]}'
```

While openAI is

```ssh
curl https://api.openai.com/v1/chat/completions \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer $OPENAI_API_KEY" \
  -d '{
     "model": "gpt-3.5-turbo",
     "messages": [{"role": "user", "content": "Say this is a test!"}],
     "temperature": 0.7
   }'
```

**Comment by D7EAD** [comment]
Hi,

Yes, this is a rather good idea for an addition. I'll consider implementing it.

**Comment by D7EAD** [comment]
Hi, again. 

I am actively working on this feature and will soon post an update here when I push the changes to a development branch for you and others to test.

Once again, thanks for the suggestion.

**Comment by D7EAD** [comment]
Hi.

This feature has been implemented in the upcoming patch #32. You can test it on branch `v3.1.0-dev` until then.

Thanks.
